frontend/src/scenes/data-warehouse/viewLinkLogic.tsx

```diff
@@ -112,44 +164,23 @@ export const viewLinkLogic = kea<viewLinkLogicType>([
     forms(({ actions, values }) => ({
         viewLink: {
             defaults: NEW_VIEW_LINK,
-            errors: ({ source_table_name, joining_table_name, joining_table_key, source_table_key }) => {
-                let joining_table_key_err: string | undefined = undefined
-                let source_table_key_err: string | undefined = undefined
-
-                if (!joining_table_key) {
-                    joining_table_key_err = 'Must select a join key'
-                }
-
-                if (!source_table_key) {
-                    source_table_key_err = 'Must select a join key'
-                }
-
-                if (
-                    joining_table_key &&
-                    source_table_key &&
-                    values.selectedJoiningTable?.columns?.find((n) => n.key == joining_table_key)?.type !==
-                        values.selectedSourceTable?.columns?.find((n) => n.key == source_table_key)?.type
-                ) {
-                    joining_table_key_err = 'Join key types must match'
-                    source_table_key_err = 'Join key types must match'
-                }
-
+            errors: ({ source_table_name, joining_table_name }) => {
                 return {
                     source_table_name: values.isNewJoin && !source_table_name ? 'Must select a table' : undefined,
                     joining_table_name: !joining_table_name ? 'Must select a table' : undefined,
-                    source_table_key: source_table_key_err,
-                    joining_table_key: joining_table_key_err,
                 }
             },
-            submit: async ({ joining_table_name, source_table_name, source_table_key, joining_table_key }) => {
+            submit: async ({ joining_table_name, source_table_name }) => {
                 if (values.joinToEdit?.id && values.selectedSourceTable) {
                     // Edit join
                     try {
                         await api.dataWarehouseViewLinks.update(values.joinToEdit.id, {
                             source_table_name: source_table_name ?? values.selectedSourceTable.name,
-                            source_table_key,
+                            source_table_key: values.selectedSourceKey ?? undefined,
+                            source_table_key_hogql: values.selectedSourceKeyHogQL ?? undefined,
                             joining_table_name,
-                            joining_table_key,
+                            joining_table_key: values.selectedJoiningKey ?? undefined,
+                            joining_table_key_hogql: values.selectedJoiningKeyHogQL ?? undefined,
```

Comment:
Maybe can reuse the non hogql join key?

Reply:
I had thought about not being able to differentiate between picking a field on a table vs a hogql expression, but I've just realized that we can do `parse_expr(selectedJoiningKey)` and it'll return a Field with a single chain, which is what we want. I'll simplify this now